src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/CSharpFlagsEnumGenerator.cs

```diff
@@ -16,14 +15,16 @@ private CSharpFlagsEnumGenerator()
     {
     }
 
+    protected override SyntaxGeneratorInternal SyntaxGenerator
+        => CSharpSyntaxGeneratorInternal.Instance;
```

Comment:
instead of the flags-generator needing to be passed the syntax-generator, it can jsut ask its subclass to provide it. t he subclass provides the internal impl (which can be used from workspace/code-fixes layer), as it supplies everything needed.